### PR TITLE
feat: add authentication prompt

### LINKED ISSUE #133: Implement the authentication prompt
## Description

Implement the authentication prompt

## Acceptance Criteria

- [x] Implement according to the design specified in https://github.com/ipfs-shipyard/pm-idm/issues/106
- [x] Integrate with `idm-wallet`

## Pull Request

https://github.com/ipfs-shipyard/nomios-web/pull/24

Related PRs:

- [Add `CheckIcon` icon](https://github.com/ipfs-shipyard/nomios-web-uikit/pull/53)
- [Add `<Drodown>` component](https://github.com/ipfs-shipyard/nomios-web-uikit/pull/62)
- [Add `animateOnEnter` prop to `<Avatar>` component](https://github.com/ipfs-shipyard/nomios-web-uikit/pull/63)
- [Add `app` object to `unlock` method](https://github.com/ipfs-shipyard/js-idm-bridge-postmsg/pull/2)

### PR DESCRIPTION
**Pull request by PedroMiguelSS**
This PR adds the authentication prompt. 

Closes [#133](https://github.com/ipfs-shipyard/pm-idm/issues/133).

⚠️It should be merged only after [this one](https://github.com/ipfs-shipyard/js-idm-bridge-postmsg/pull/2). Dependencies must be updated.

### CODE REVIEW

File: src/app-mediator/screens/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import Pristine from './pristine';
```
Comment (satazor) [inline src/app-mediator/screens/index.js]: Can we directly export?

File: src/pages/authenticate/credentialScopes.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { UserIcon, CheckIcon } from '@nomios/web-uikit';
```
Comment (satazor) [inline src/pages/authenticate/credentialScopes.js]: Question: should this go to the [data](https://github.com/ipfs-shipyard/nomios-web/tree/master/src/shared/data) folder? No strong opinion, just asking.

File: src/pages/authenticate/identity-selector/IdentitySelector.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        if (value !== 'createCta') {
+            this.selectedValue = value;
+            this.props.onChange && this.props.onChange(value);
+        }
+    };
+}
+
+IdentitySelector.propTypes = {
+    identities: PropTypes.array.isRequired,
+    initialValueId: PropTypes.string,
```
Comment (satazor) [inline src/pages/authenticate/identity-selector/IdentitySelector.js]: Shouldn't this be `initialIdentityId`?

File: src/pages/authenticate/identity-selector/IdentitySelector2.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
```
Comment (satazor) [inline src/pages/authenticate/identity-selector/IdentitySelector2.js]: Why do we have a `IdentitySelector2` file?

**Review by satazor — CHANGES_REQUESTED** [review]

File: src/app-mediator/screens/index.js (hunk 1 above)
Comment (PedroMiguelSS) [reply src/app-mediator/screens/index.js]: Sure, I'll keep it consistent 👍

File: src/pages/authenticate/credentialScopes.js (hunk 2 above)
Comment (PedroMiguelSS) [reply src/pages/authenticate/credentialScopes.js]: No strong opinion as well. Moved it already to `data` folder.

File: src/pages/authenticate/identity-selector/IdentitySelector2.js (hunk 4 above)
Comment (PedroMiguelSS) [reply src/pages/authenticate/identity-selector/IdentitySelector2.js]: Forgot to delete it 🤦‍♂️

**Comment by satazor** [comment]
@PedroMiguelSS can you fix the conflicts?

**Comment by PedroMiguelSS** [comment]
You've merged your PR and then my PR got conflicts 😂 I'll take care of it, again 😂

**Review by satazor — APPROVED** [review]

**Pull request merged**